src/main/java/org/trips/service_framework/notificationHandler/SMSNotificationHandler.java

```diff
@@ -0,0 +1,31 @@
+package org.trips.service_framework.notificationHandler;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.trips.service_framework.clients.MercuryClient;
+import org.trips.service_framework.clients.request.NotificationRequest;
+import org.trips.service_framework.clients.response.NotificationResponse;
+import org.trips.service_framework.dtos.SMSData;
+import org.trips.service_framework.utils.Context;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SMSNotificationHandler implements NotificationHandler {
+    private final MercuryClient mercuryClient;
+
+    @Override
+    public <T> NotificationResponse send(String subject, String clientCode, T data) {
+        NotificationRequest<SMSData> request = NotificationRequest.<SMSData>builder()
+                .user(Context.getUserId())
+                .clientCode(clientCode)
+                .subject(subject)
+                .data((SMSData) data)
+                .build();
+        NotificationResponse response = mercuryClient.sendSMS(request);
```

Comment:
in case of error, the notification handler should not throw error rather return with a response that there was an error sending a notification. We can't let the whole flow fail because of a failure in sending a notification. Implementing services can have a check basis the response if they want to throw error or not.

looping in @anomitra @dongiveajack for their views.